sdk/containerregistry/azure-containers-containerregistry/src/main/java/com/azure/containers/containerregistry/ContainerRepositoryAsyncClient.java

```diff
@@ -0,0 +1,650 @@
+            .serializerAdapter(serializerAdapter)
+            .url(endpoint).buildClient();
+
+        this.endpoint = endpoint;
+        this.repositoryName = repositoryName;
+        this.registriesImplClient = registryImpl.getContainerRegistries();
+        this.serviceClient = registryImpl.getContainerRegistryRepositories();
+        this.apiVersion = apiVersion;
+    }
+
+    /**
+     * Get endpoint associated with the class.
+     * @return String the endpoint associated with this client.
+     */
+    public String getEndpoint() {
+        return this.endpoint;
+    }
+
+
+    /**
+     * Get the registry associated with the client.
+     * @return Return the registry name.
+     */
+    public String getRegistry() {
+        return this.endpoint;
+    }
+
+    /**
+     * Get repository associated with the class.
+     * @return Return the repository name.
+     * */
+    public String getRepository() {
+        return this.repositoryName;
+    }
+
+    /**
+     * Delete repository.
+     *
+     * @throws ClientAuthenticationException thrown if the client's credentials do not have access to modify the namespace.
+     * @return the completion.
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public Mono<Response<DeleteRepositoryResult>> deleteWithResponse() {
+        return withContext(context -> deleteWithResponse(context));
+    }
+
+    /**
+     * Delete repository.
+     *
+     * @param context Context associated with the operation.
+     * @throws ClientAuthenticationException thrown if the client's credentials do not have access to modify the namespace.
+     * @throws RuntimeException all other wrapped checked exceptions if the request fails to be sent.
+     * @return the completion.
+     */
+    Mono<Response<DeleteRepositoryResult>> deleteWithResponse(Context context) {
+        try {
+            if (context == null) {
+                return monoError(logger, new NullPointerException("'context' cannot be null."));
+            }
+
+            return this.registriesImplClient.deleteRepositoryWithResponseAsync(repositoryName, context)
+                .map(res -> Utils.mapResponse(res, Utils::mapDeleteRepositoryResult))
+                .onErrorMap(Utils::mapException);
+        } catch (RuntimeException ex) {
+            return monoError(logger, ex);
+        }
+    }
+
+    /**
+     * Delete repository.
+     *
+     * @throws IllegalArgumentException thrown if parameters fail the validation.
+     * @throws ClientAuthenticationException thrown if the client's credentials do not have access to modify the namespace.
+     * @return the completion.
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public Mono<DeleteRepositoryResult> delete() {
+        return this.deleteWithResponse().map(Response::getValue);
+    }
+
+    /**
+     * Delete registry artifact.
+     *
+     * @param digest Digest name.
+     * @throws IllegalArgumentException thrown if parameters fail the validation.
```

Comment:
Let's call out what parameters are invalid

```suggestion
     * @throws IllegalArgumentException If {@code digest} is null.
```

Reply:
Actually, looks like this is never thrown

Reply:
Yup as of now there is no validation of the arguments other than null, will remove it from other places too.

---
In reply to: [609152473](https://github.com/Azure/azure-sdk-for-java/pull/20368#discussion_r609152473) [](ancestors = 609152473)